### PR TITLE
Explicitly require Java 8 compatibility in the library

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This is a copy of some bits in Jenkins Plugin Pom. It is a follow-up to the discussion with @jglick in https://github.com/kohsuke/access-modifier/pull/4

- [x] Make Java requirement explicit in Maven Compiler. Stick to Java 6 for a while
- [x] Enable enforcer for dependency checks

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,6 +15,8 @@
   <properties>
     <findbugs-maven-plugin.version>3.0.4</findbugs-maven-plugin.version>
     <findbugs.failOnError>true</findbugs.failOnError>
+    <java.level>6</java.level>
+    <java.level.test>6</java.level.test>
```
Comment (jglick) [inline pom.xml]: Probably better to default to `${java.level}`; or simply omit, as differentiating them currently does not work in IDEA so other developers would complain.

**Comment by oleg-nenashev** [comment]
@segrey would you be fine if I set Java 8 as a new requirement? It is fine for our projects, but I am not sure about yours

**Comment by segrey** [comment]
@oleg-nenashev Yes, it's OK for IntelliJ IDEs too.

**Comment by oleg-nenashev** [comment]
```
Failed to execute goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check (check) on project winp: Execution check of goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.15:check failed. NullPointerException
```

This is something I know how to fix. Will finalize it and merge

**Comment by mikecirioli** [comment]
We probably need to be looking at Java 11 compatibility at this point (see [this JEP PR](https://github.com/jenkinsci/jep/pull/382))

**Pull request closed without merging**